docs/spec/exceptions.rst

```diff
@@ -0,0 +1,42 @@
+Exceptions
+==========
+
+Some type checking behaviors, such as type narrowing and reachability analysis,
```

Comment:
I'm not sure if this introductory section is needed for context. I could be convinced to simply drop it for brevity.

Reply:
I find the concrete example helpful, as a newcomer.

I also wouldn't mind an example for the ContextManager section; I had to look up the original mypy bug report to see what this was addressing.

Maybe something like

> For example, type checkers that employ code flow analysis will be able to identify whether `return default` is reachable, depending on whether `context.__exit__` "suppresses" exceptions:
> ```
> with context():
>     return d["foo"]
> return default
> ```